File: vs2003_cd01/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/Samples/Tutorials/clr_fundamentals/samples/typemembers/properties/vb/Properties.vb
Imports System

Class Employee
   Private _name As String
   Private _age As Int32

   Public Property Name As String
      Get 
         Name = _name
      End Get
      Set 
         _name = Value
      End Set
   End Property

   Public Property Age As Int32
      Get 
         Age = _age
      End Get
      Set  
         If value < 0 Then _
            Throw New ArgumentOutOfRangeException("Age must be >= 0")
         _age = Value
      End Set
   End Property
End Class


Class App 
   Shared Sub Main()
      Dim employeeVar As New Employee()
      employeeVar.Name = "Fred Flintstone"
      employeeVar.Age = 30
      Try
         employeeVar.Age = -5
      Catch e As Exception
         Console.WriteLine("Failed to set age: " + e.Message + ".")
      End Try

      Console.Write("Press Enter to close window...")
      Console.Read()
   End Sub
End Class

File: vs2003_cd01/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/Samples/Tutorials/clr_fundamentals/samples/typemembers/properties/cs/Properties.cs
using System;

class Employee {
   private String _name;
   private Int32 _age;

   public String Name {
      get { return _name; }
      set { _name = value; }
   }

   public Int32 Age {
      get { return _age; }
      set { 
         if (value < 0) 
            throw new ArgumentOutOfRangeException("Age must be >= 0");
         _age = value;
      }
   }
}


class App {
   static void Main() {
      Employee employee = new Employee();
      employee.Name = "Fred Flintstone";
      employee.Age = 30;
      try {
         employee.Age = -5;
      }
      catch (Exception e) {
         Console.WriteLine("Failed to set age: " + e.Message + ".");
      }

      Console.Write("Press Enter to close window...");
      Console.Read();
   }
}
